camelot/view/action_steps/open_file.py

```diff
@@ -43,21 +48,44 @@ class OpenFile( ActionStep, DataclassSerializable ):
     :param file_name: the absolute path to the file to open
     
     The :keyword:`yield` statement will return :const:`True` if the file was
-    opend successfull.
+    opened successfully.
     """
 
-    path: str
+    path: InitVar[str]
+    type: str="websocket" # "content" or "url" or "websocket"
+
+    url: str = field(init=False, default=None)
+    content: str = field(init=False, default=None)
+    filename: str = field(init=False, default=None)
+
     blocking: bool = False
 
+    def __post_init__(self, path):
```

Comment:
@tim-vdm : not sure hoe jij dit ziet, maar ik zou geneigd zijn de 'gebruiker vd action step' het type niet te laten kiezen,
anders gaan we altijd op verschillende plaatsen moeten gaan evalueren hoe we dat willen doen.

ik zou dat eerder ofwel globaal, ofwel door de action step zelf laten beslissen

bvb, klein bestand : content, heel groot, url  of zo